### PR TITLE
Scenetransition, Questsystem and other fixes

### PR DESCRIPTION
**Pull request by duriumm**
Started out as quest branch but ended up being scene transition fixes and other small stuff

- Basic quest system working by killing x amount of enemies chosen in inspector npc window
- Chest sprite that can be opened and looted (Bug remains where opening sound is played on looting)
- Custom colliders for water/grass edge tiles 
- Standing lantern with flickering light 
- Trader NPC now uses the interract script on key press E, which should be used for all further interractions
- Individual DontDestroyOnLoad scripts added to Player, Main camera and CM vcam1 
- Player can no longer attack while shop is open
- Walking in/out of houses enabled with new scenetransition object where two scenetransition objects always are linked to eachother by gameobject name

### CODE REVIEW

File: Pixel Game/Assets/Scripts/Enemy scripts/EnemyHealth.cs [hunk 1]
```diff
@@ -28,6 +28,12 @@ protected override void Start()
 
     protected override void Kill()
     {
+
+        if(dataToPassBetweenScenes.currentActivePlayerQuest.questType == Quest.QUESTTYPE.KILL_ENEMIES)
+        {
+            dataToPassBetweenScenes.currentActivePlayerQuest.IncrementKilledEnemies();
+        }
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/Enemy scripts/EnemyHealth.cs]: Did you intend to keep this? `dataToPassBetweenScenes` is not declared so there's a compiler error.

Comment (duriumm) [reply Pixel Game/Assets/Scripts/Enemy scripts/EnemyHealth.cs]: Yes I intend to keep it. ```dataToPassBetweenScenes``` gameObject needs to be redeclared in start()

File: Pixel Game/Assets/Scripts/FlickerLamp.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FlickerLamp : MonoBehaviour
+{
+    UnityEngine.Experimental.Rendering.Universal.Light2D torch2dLightComponent;
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/FlickerLamp.cs]: I would suggest putting a
``` c#
using UnityEngine.Experimental.Rendering.Universal;
```
at the top of the file and then you just have to type `Light2D` when referring to it:
```c#
Light2D torch2dLightComponent;
```

File: Pixel Game/Assets/Scripts/FlickerLamp.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    private float maxIntensity;
+
+
+
+
+    void Start()
+    {
+
+        //fireParticleEffect = this.gameObject.GetComponent<ParticleSystem>();
+        torch2dLightComponent = this.gameObject.GetComponent<UnityEngine.Experimental.Rendering.Universal.Light2D>();
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/FlickerLamp.cs]: and:
``` c#
torch2dLightComponent = this.gameObject.GetComponent<Light2D>();
```

File: Pixel Game/Assets/Scripts/SceneTransition.cs [hunk 4]
```diff
@@ -12,18 +12,29 @@ public class SceneTransition : MonoBehaviour
 
     public Animator transition;
 
+    private string transitionId;
+
+    private Vector3 testSpawnPos;
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/SceneTransition.cs]: Maybe change name to just `spawnPos`?

File: Pixel Game/Assets/Scripts/TraderNpc.cs [hunk 5]
```diff
@@ -66,4 +67,18 @@ private void OnTriggerExit2D(Collider2D collision)
...
+    public void OpenNpcShop()
+    {
+        if (shopScreen.GetComponent<CanvasGroup>().alpha == 1) // If shopscreen is visible
+        {
+            shopScreen.GetComponent<ShopScreen>().CloseShopScreen();
+        }
+        else if (shopScreen.GetComponent<CanvasGroup>().alpha == 0) // If shopscreen is NOT visible
+        { 
+            shopScreen.GetComponent<ShopScreen>().OpenShopScreen();
+        }
```
Comment (yousernaym) [inline Pixel Game/Assets/Scripts/TraderNpc.cs]: A shorter way of writing would be to just check alpha once and then an empty else:
```c#
if (alpha > 0)
    close shop
else
    open shop
```

**Review by yousernaym — COMMENTED** [review]
I suggested some minor changes but overall I think it looks very nice.

File: Pixel Game/Assets/Scripts/SceneTransition.cs (hunk 4 above)
Comment (duriumm) [reply Pixel Game/Assets/Scripts/SceneTransition.cs]: Yes of course!

File: Pixel Game/Assets/Scripts/TraderNpc.cs (hunk 5 above)
Comment (duriumm) [reply Pixel Game/Assets/Scripts/TraderNpc.cs]: Nice input!

**Comment by yousernaym** [comment]
Perfect!

**Pull request merged**